src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -371,6 +385,9 @@ private ImmutableArray<TypeParameterSymbol> MakeTypeParameters(DiagnosticBag dia
                         ImmutableArray.Create(location),
                         ImmutableArray.Create(parameter.GetReference()));
 
+                // Force attribute binding for diagnostics
+                typeParameter.GetAttributesBag(diagnostics);
```

Comment:
With my second PR they will go to the same bag -- the one that is passed to `GrabDiagnostics`. I think that makes sense for consistency.